### PR TITLE
[#IOCIT-34] Upsert Activation for PN courtesy service

### PR DESCRIPTION
**Pull request by BurnedMarshal**
#### List of Changes

New Endpoint to update the Activation status for Piattaforma Notifiche special service.
New env configuration requirements
New PN service and controller

#### Motivation and Context

The _Piattaforma Notifiche Avvisi di cortesia_ special service need two endpoint to read the current activation status and update the value. This PR introduce the new endpoint for the update of the activation status with the infrastructure to configure all the params and services for both the required APIs.

#### How Has This Been Tested?

~not tested yet~
unit test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] Unit test

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -462,6 +463,64 @@ export const EUCOVIDCERT_API_CLIENT = EUCovidCertAPIClient(
...
+    return process.exit(1);
+  })
+);
+
+export const PN_ADDRESS_BOOK_CLIENT_SELECTOR: O.Option<ReturnType<
+  typeof PNClientFactory
+>> = pipe(
+  PNAddressBookConfig,
+  E.fromPredicate(IEnabledPnAddressBookConfig.is, () => O.none),
+  E.map(_ =>
```
Comment (michaeldisaro) [inline src/config.ts:510]: Can we give it a name?

File: src/controllers/__tests__/cgnOperatorSearchController.test.ts [hunk 2]
```diff
@@ -19,8 +19,14 @@ import {
   OrderingEnum
 } from "../../../generated/cgn-operator-search/OfflineMerchantSearchRequest";
 import { NonNegativeInteger } from "@pagopa/ts-commons/lib/numbers";
-import { CardActivated } from "../../../generated/io-cgn-api/CardActivated";
-import { CardExpired } from "../../../generated/io-cgn-api/CardExpired";
+import {
```
Comment (michaeldisaro) [inline src/controllers/__tests__/cgnOperatorSearchController.test.ts:22]: These changes should not be in this PR.

**Review by michaeldisaro — CHANGES_REQUESTED** [review]
Are we sure we want some CGN related changes in this PR?

File: api_cgn.yaml [hunk 3]
```diff
@@ -242,6 +242,8 @@ definitions:
     $ref: "https://raw.githubusercontent.com/pagopa/io-functions-cgn/v2.0.0/openapi/index.yaml#/definitions/CommonCard"
   CardPending:
     $ref: "https://raw.githubusercontent.com/pagopa/io-functions-cgn/v2.0.0/openapi/index.yaml#/definitions/CardPending"
+  CardPendingDelete:
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply api_cgn.yaml]: Moved to another PR (#918)

File: src/controllers/__tests__/cgnOperatorSearchController.test.ts (hunk 2 above)
Comment (BurnedMarshal) [reply src/controllers/__tests__/cgnOperatorSearchController.test.ts]: Moved to another PR (https://github.com/pagopa/io-backend/pull/918)

**Review by michaeldisaro — APPROVED** [review]
lgtm

**Pull request merged**